Pause menu

awktopus1 opened the pull request:
- Added pause menu (#6)
 - Added save / load screen

 - Need to add settings menu
 - Need to add functionality to exit buttons
 - Needs more styling/theming

awktopus1 committed:
Pause Update

 - Added "pause" to input map with key "Escape"
 - Added background to darken screen when paused
 - Added panel for pause menu items
 - Added pause code

awktopus1 committed:
Added buttons & theme

 - Added 3 new fonts to choose from
 - Added new theme with basic buttons
 - Added buttons and label to pause panel

awktopus1 committed:
Pause Menu and Save Popup

 - Added buttons
 - Modified theme
 - Added new panel for saving

awktopus1 committed:
Reorganised UI Structure

 - Rewrote cude for handling UI, functionality mostly unchanged
 - Bug when using Esc while save screen open

awktopus1 committed:
New Save/Load Screen

 - Added save/load screen with list of saves, and options to add a new save, load previous saves and overwrite previous saves. There is also a rich text area for save info.

awktopus1 committed:
Reorganised file structure and added save panel features

 - merged save and load buttons
 - added confirmation popup to overwrite saves
 - added list of saves (wip) that you can add to

awktopus1 committed:
Save screen delete button

awktopus1 committed:
Updated methods and rewrote save list

Aurora2500 commented on galaxy/SavePanel.gd:
```diff
@@ -0,0 +1,69 @@
+		get_parent().confirmation_popup("Are you sure you want to overwrite this save?", self)
+			
+	
+func _on_DeleteButton_pressed():
+	button_ = "D"
+	if current_selection > -1:
+		get_parent().confirmation_popup("Are you sure you want to delete this save?", self)
```
What's with the `button_ = "D"`?
If that's for the user to press. I'd say it's better to add it as an input map under `Project > Project settings... > Input Map`, that way it can be easily changed when we need to. Plus anyone will be able to set up their own keymaps.

awktopus1 replied:
That's to do with remembering which popup is open. Because popups are handled by the main UI code, i need to store which one i requested so when it comes back i know what action to take. D stand for Delete and the other option is S for save overwriting.

Aurora2500 replied:
Makes sense.
In that case:
1. Seems to me like it would make more sense to use an enum.
2. Maybe have a more self explanatory name for the variable, like `current_action` or whatever else you see fit.

Aurora2500 commented on galaxy/SavePanel.gd:
```diff
@@ -0,0 +1,69 @@
+		get_parent().confirmation_popup("Are you sure you want to delete this save?", self)
+		
+
+func handle_confirm():
+	if button_ == "D":
+		delete_save(current_selection)
+	get_parent().close_confirm_popup()
+
```
Same thing as in `_on_DeleteButton_pressed`

Aurora2500 commented on galaxy/SavePanel.gd:
```diff
@@ -0,0 +1,69 @@
+		add_new_save(new_save_name)
+
+
+func _on_OverwriteSaveButton_pressed():
+	button_ = "S"
+	if current_selection > -1:
+		get_parent().confirmation_popup("Are you sure you want to overwrite this save?", self)
+			
```
same thing as `_onDeleteButton_Pressed`

awktopus1 committed:
Fixed bug in save deleting and refactored saving code

 - changed button_ to current_action, using actions enum
 - fixed deleted saves reappearing after refreshing

Aurora2500 reviewed (approved):
LGTM

Aurora2500 committed:
moved MainScene to try and help with the merge conflict

Aurora2500 committed:
merged development into PauseMenu so that the PR can be successfull
